include/internal/string.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 1995-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_INTERNAL_STRING_H
+# define HEADER_INTERNAL_STRING_H
+
+# include <string.h>
+# include <malloc.h>
+
+# ifdef OPENSSL_SYS_WINDOWS
```

Comment:
I still argue that we are better off using compiler pre-defines. Current consensus seems to be at least to be consistent within same file. And in such case it would be natural to use pre-defines in *new* files. _WIN32 is the one to use in such situation.

Reply:
Besides, in this particular case you actually introduce header position dependency. Because you rely on header that defines OPENSSL_SYS_WINDOWS to be included prior this one.